### ISSUE TITLE
training qwen3-vl-4B or 2B model very slow when using torch2.9

### ISSUE THREAD

**Issue by pinecypressfxd** [QwenLM/Qwen3-VL#1701]
[2025-10-27 16:11:09,403] [WARNING] [lr_schedules.py:858:get_lr] Attempting to get learning rate from scheduler before it has started
  0%|                                                                                                                        | 0/300000 [00:00<?, ?it/s][2025-10-27 16:11:09,495] [WARNING] [lr_schedules.py:858:get_lr] Attempting to get learning rate from scheduler before it has started
  0%|                                                                                                                        | 0/300000 [00:00<?, ?it/s][2025-10-27 16:11:09,595] [WARNING] [lr_schedules.py:858:get_lr] Attempting to get learning rate from scheduler before it has started
  0%|                                                                                                                        | 0/300000 [00:00<?, ?it/s][2025-10-27 16:11:10,360] [WARNING] [lr_schedules.py:858:get_lr] Attempting to get learning rate from scheduler before it has started
loss:0.8416:   0%|                                                                                           | 15/loss:0.8406:   0%|                                                     | 15/300000 [43:28<13547:29:44, 162.58s/it]loss:0.8406:   0%|                                                     | 16/loss:0.8096:   0%|                                loss:0.78:   0%|                                                                                                                      | 18/300000 [48:53<13546:07:08, 162.56s/it]

**Comment by John-Ge** [comment]
You could try to use pytorch 2.6.

**Comment by olliepro** [comment]
I was seeing times 30x longer on torch==2.9.0. Issue fixed by using torch<2.9

**Comment by dillonalaird** [comment]
Had the same issue. Pytorch 2.9 I was getting 400s per iteration for my training run but with 2.8 it went down to 30s per iteration.

**Comment by pinecypressfxd** [comment]
> You could try to use pytorch 2.6.

thank you for reply. I intended to use the muon optimizer in torch 2.9 to train the model. Later I swithched back to qwen2.5-vl for testing the muon in torch2.9.
